File: MAP/lab7/lab7/Models/PrintStmt.cs
using System;
using System.Collections;
using lab7.Utils;

namespace lab7.Models
{
    public class PrintStmt : IStatement
    {
        private IExpression exp;

        public PrintStmt(IExpression exp)
        {
            this.exp = exp;
        }

        public PrgState execute(PrgState state)
        {
  
            int result = exp.eval(state.Dict);          
            state.OutList.add(result);
            
            return state;
        }

        public override string ToString()
        {
            return "Print(" + exp + ")";
        }
    }
}

File: MAP/lab8 - ProcTable/src/models/statement/PrintStmt.java
package models.statement;

import models.PrgState;
import models.expression.IExpression;
import utils.IDictionary;
import utils.IHeap;
import utils.IList;

public class PrintStmt implements Statement {

    private IExpression exp;
    public PrintStmt(IExpression _exp){
        exp = _exp;
    }


    @Override
    public PrgState execute(PrgState state) {
        IList<Integer> list = state.getList();
        IDictionary<String, Integer> dict = state.peekStackSymTab();
        IHeap<Integer, Integer> heap = state.getHeap();
        int val = exp.eval(dict, heap);
        list.add(val);
        return null;
    }

    @Override
    public String toString(){ return "print(" + exp + ");" ;}
}
